modules/virtualisation/oci-containers.nix

```diff
@@ -0,0 +1,31 @@
+# Equivalent of https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/virtualisation/oci-containers.nix
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.virtualisation.oci-containers;
+
+  inherit (lib) mkDefault mkIf mkMerge mkOption types;
+
+  defaultBackend = "podman";
+in {
+
+  meta.maintainers = [ pkgs.lib.maintainers.michaelCTS ];
+  options.virtualisation.oci-containers = {
+
+    enable = lib.mkEnableOption
+      "a convenience option to enable containers in platform-agnostic manner";
```

Comment:
This is taken from `nixpkgs` https://github.com/NixOS/nixpkgs/blob/ec68439bdc77ee684a4b2846f03677c5c50920a9/nixos/modules/virtualisation/oci-containers.nix#L335-L349

The problem with `containers` is that it used by `podman`. I'm not versed enough in nix and nixpkgs to not create a recursion problem with `containers.enable` using `containers.backend` to enable `podman` which itself then sets `containers.enable` and configures the settings and other stuff. I gave it a shot and immediately ran into recursion errors.

Since `nixpkgs` doesn't bother with that, I erred on the side of "they probably know better what they're doing that some 1 year nix scrub like me".